stdlib/public/core/Array.swift

```diff
@@ -1865,3 +1869,189 @@ internal struct _ArrayAnyHashableBox<Element: Hashable>
     return true
   }
 }
+
+// SWIFT_ENABLE_TENSORFLOW
+extension Array where Element : Differentiable {
+  /// Views the array as the differentiable product manifold of `Element` multiplied with itself
+  /// `count` times.
+  @_fixed_layout
+  public struct DifferentiableView : Differentiable {
+    private var base: [Element]
+
+    /// The viewed array.
+    // I'm implementing this as a computed property instead of directly exposing `base` because the
+    // `@differentiable` annotation does not make the stored property actually differentiable. I
+    // think this is a bug. Maybe it's related to `@_fixed_layout`?
+    // TODO: Determine if that is a bug, and fix.
+    public var array: [Element] {
+      @differentiable(wrt: self, vjp: _vjpArray)
+      get {
+        return base
+      }
+      set(newValue) {
+        base = newValue
+      }
+    }
+
+    public func _vjpArray() -> ([Element], (Array<Element>.CotangentVector) -> CotangentVector) {
+      return (array, { $0 })
+    }
+
+    /// Construct a view of the given array.
+    @differentiable(wrt: array, vjp: _vjpInit)
+    public init(_ array: [Element]) { self.base = array }
+
+    public static func _vjpInit(_ array: [Element]) ->
+      (Array.DifferentiableView, (CotangentVector) -> CotangentVector)
+    {
+      return (Array.DifferentiableView(array), { $0 })
+    }
+
+    // MARK: - Differentiable conformance.
+
+    public typealias TangentVector = Array<Element.TangentVector>.DifferentiableView
+    public typealias CotangentVector = Array<Element.CotangentVector>.DifferentiableView
+    public typealias AllDifferentiableVariables =
+      Array<Element.AllDifferentiableVariables>.DifferentiableView
+
+    public var allDifferentiableVariables: AllDifferentiableVariables {
+      get {
+        return AllDifferentiableVariables(array.map { $0.allDifferentiableVariables })
+      }
+      set(newValue) {
```

Comment:
Omit `newValue` since the name is implicitly bound.
```suggestion
      set {
```

Reply:
done